dkpro-core-io-bincas-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/bincas/BinaryCasReader.java

```diff
@@ -144,12 +161,24 @@ public void getNext(CAS aCAS)
 			DataInputStream dis = new DataInputStream(bis);
 			dis.read(header);
 			
-			// Use externalized type system
-			if (typeSystemLocation != null) {
-				xts = readTypeSystem(header,aCAS);				
-			}
-			
-			xts = readDKProHeader(bis, header, xts);
+ 			// If it is DKPro Core format, read the type system
+			if (Arrays.equals(header, DKPRO_HEADER)) {
+				xts = readDKProHeader(bis, header, xts);
+	 		} else {
+	 			// No embedded DKPro TS, reset
+	 			try {
+	 				bis.reset();
+	 			} catch (IOException e) {
+	 				throw new IOException(e);
```

Comment:
Catching an exception and then throwing the same type of exception is redundant. Just don't try/catch.